version-catalog/src/functionalTest/kotlin/io/spine/internal/catalog/SpineVersionCatalogFunctionalTest.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.catalog
+
+import java.nio.file.Files
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+
+/**
+ * Verifies the generated type-safe accessors for `Dummy` dependency.
+ *
+ * `Dummy` dependency is an imaginary library, which exists for two reasons:
+ *
+ *  1. Showcasing API for dependency declarations.
+ *  2. Functional testing in conditions, which are very close to real life.
+ *
+ * Actually, our `SpineVersionCatalog` provides code upon Gradle-provided
+ * `VersionCatalogBuilder`. This code can fill up the passed catalog. And, as
+ * for now, there's no any other legitimate way (except for a true functional test)
+ * to check if this code executes successfully on a real instance of `VersionCatalogBuilder`.
+ *
+ * See issue in Gradle: https://github.com/gradle/gradle/issues/20807
+ *
+ * A note about `FunctionName` suppression. In JUnit in Kotlin, it is okay
+ * to write test names just in function name. But IDEA does not report it as en
+ * error only for `test` source set. It's unclear how to disable it for custom
+ * source sets.
+ */
+@Suppress("FunctionName") // See docs above.
+@DisplayName("`SpineVersionCatalog` should")
+class SpineVersionCatalogFunctionalTest {
+
+    private val projectDir = Files.createTempDirectory(this::class.simpleName).toFile()
```

Comment:
@yevhenii-nadtochii, when do we delete this temp. dir?

Reply:
I believed it is deleted automatically on application exit. But it's not true, it is left to the mercy of OS.

[Baeldung](https://www.baeldung.com/java-temp-directories) states:

>> One of the most popular approaches used to create temporary directories is to delegate the destination to the underlying operating system. The location is given by the java.io.tmpdir property, and every operating system has its own structure and cleanup routines.

I've fixed this. Now, it is deleted explicitly on the test completion.